`TestScheduler` is flaky

## Flaky Test

### Which jobs are failing
```
--- FAIL: TestScheduler (6.34s)
    scheduler_test.go:73: 
        	Error Trace:	scheduler_test.go:73
        	            				scheduler_test.go:106
        	            				scheduler_test.go:133
        	Error:      	Received unexpected error:
        	            	invalid character '"' after top-level value
        	Test:       	TestScheduler
```

### CI link
https://github.com/tikv/pd/actions/runs/3216756484/jobs/5258969812

### Reason for failure (if possible)


### Anything else

--- a/server/handler.go
+++ b/server/handler.go
@@ -988,7 +988,7 @@ func (h *Handler) SetStoreLimitTTL(data string, value float64, ttl time.Duration
 	}, ttl)
 }
 
-// IsLeader return ture if this server is leader
+// IsLeader return true if this server is leader
 func (h *Handler) IsLeader() bool {
 	return h.s.member.IsLeader()
 }

--- a/server/schedule/checker/rule_checker.go
+++ b/server/schedule/checker/rule_checker.go
@@ -379,7 +379,7 @@ func (c *RuleChecker) strategy(region *core.RegionInfo, rule *placement.Rule) *R
 		isolationLevel: rule.IsolationLevel,
 		locationLabels: rule.LocationLabels,
 		region:         region,
-		extraFilters:   []filter.Filter{filter.NewLabelConstaintFilter(c.name, rule.LabelConstraints)},
+		extraFilters:   []filter.Filter{filter.NewLabelConstraintFilter(c.name, rule.LabelConstraints)},
 	}
 }
 

--- a/server/schedule/filter/filters.go
+++ b/server/schedule/filter/filters.go
@@ -518,8 +518,8 @@ type labelConstraintFilter struct {
 	constraints []placement.LabelConstraint
 }
 
-// NewLabelConstaintFilter creates a filter that selects stores satisfy the constraints.
-func NewLabelConstaintFilter(scope string, constraints []placement.LabelConstraint) Filter {
+// NewLabelConstraintFilter creates a filter that selects stores satisfy the constraints.
+func NewLabelConstraintFilter(scope string, constraints []placement.LabelConstraint) Filter {
 	return labelConstraintFilter{scope: scope, constraints: constraints}
 }
 

--- a/server/schedule/filter/filters_test.go
+++ b/server/schedule/filter/filters_test.go
@@ -89,7 +89,7 @@ func TestLabelConstraintsFilter(t *testing.T) {
 		{"_id", "notExists", []string{}, plan.StatusOK},
 	}
 	for _, testCase := range testCases {
-		filter := NewLabelConstaintFilter("", []placement.LabelConstraint{{Key: testCase.key, Op: placement.LabelConstraintOp(testCase.op), Values: testCase.values}})
+		filter := NewLabelConstraintFilter("", []placement.LabelConstraint{{Key: testCase.key, Op: placement.LabelConstraintOp(testCase.op), Values: testCase.values}})
 		re.Equal(testCase.res, filter.Source(testCluster.GetOpts(), store).StatusCode)
 	}
 }

--- a/tests/pdctl/scheduler/scheduler_test.go
+++ b/tests/pdctl/scheduler/scheduler_test.go
@@ -22,6 +22,7 @@ import (
 
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/stretchr/testify/require"
+	"github.com/tikv/pd/pkg/testutil"
 	"github.com/tikv/pd/server/config"
 	"github.com/tikv/pd/server/versioninfo"
 	"github.com/tikv/pd/tests"
@@ -74,6 +75,15 @@ func TestScheduler(t *testing.T) {
 		return ""
 	}
 
+	mightExec := func(args []string, v interface{}) {
+		output, err := pdctl.ExecuteCommand(cmd, args...)
+		re.NoError(err)
+		if v == nil {
+			return
+		}
+		json.Unmarshal(output, v)
+	}
+
 	mustUsage := func(args []string) {
 		output, err := pdctl.ExecuteCommand(cmd, args...)
 		re.NoError(err)
@@ -100,10 +110,13 @@ func TestScheduler(t *testing.T) {
 		re.Equal(expectedConfig, configInfo)
 	}
 
-	checkSchedulerDescibeCommand := func(schedulerName, expectedStatus, expectedSummary string) {
-		time.Sleep(time.Millisecond * 50)
+	checkSchedulerDescribeCommand := func(schedulerName, expectedStatus, expectedSummary string) {
 		result := make(map[string]interface{})
-		mustExec([]string{"-u", pdAddr, "scheduler", "describe", schedulerName}, &result)
+		testutil.Eventually(re, func() bool {
+			mightExec([]string{"-u", pdAddr, "scheduler", "describe", schedulerName}, &result)
+			return len(result) != 0
+		}, testutil.WithTickInterval(50*time.Millisecond))
+
 		re.Equal(expectedStatus, result["status"])
 		re.Equal(expectedSummary, result["summary"])
 	}
@@ -130,7 +143,7 @@ func TestScheduler(t *testing.T) {
 
 	echo := mustExec([]string{"-u", pdAddr, "config", "set", "enable-diagnostic", "true"}, nil)
 	re.Contains(echo, "Success!")
-	checkSchedulerDescibeCommand("balance-region-scheduler", "pending", "1 store(s) RegionNotMatchRule; ")
+	checkSchedulerDescribeCommand("balance-region-scheduler", "pending", "1 store(s) RegionNotMatchRule; ")
 
 	// scheduler delete command
 	args := []string{"-u", pdAddr, "scheduler", "remove", "balance-region-scheduler"}
@@ -141,7 +154,7 @@ func TestScheduler(t *testing.T) {
 	}
 	checkSchedulerCommand(args, expected)
 
-	checkSchedulerDescibeCommand("balance-region-scheduler", "disabled", "")
+	checkSchedulerDescribeCommand("balance-region-scheduler", "disabled", "")
 
 	schedulers := []string{"evict-leader-scheduler", "grant-leader-scheduler"}
 
@@ -184,7 +197,6 @@ func TestScheduler(t *testing.T) {
 		}
 		checkSchedulerCommand(args, expected)
 
-		// check the compactily
 		// scheduler add command
 		args = []string{"-u", pdAddr, "scheduler", "add", schedulers[idx], "2"}
 		expected = map[string]bool{
@@ -421,12 +433,12 @@ func TestScheduler(t *testing.T) {
 	checkSchedulerWithStatusCommand(nil, "paused", []string{
 		"balance-leader-scheduler",
 	})
-	checkSchedulerDescibeCommand("balance-leader-scheduler", "paused", "")
+	checkSchedulerDescribeCommand("balance-leader-scheduler", "paused", "")
 
 	mustUsage([]string{"-u", pdAddr, "scheduler", "resume", "balance-leader-scheduler", "60"})
 	mustExec([]string{"-u", pdAddr, "scheduler", "resume", "balance-leader-scheduler"}, nil)
 	checkSchedulerWithStatusCommand(nil, "paused", nil)
-	checkSchedulerDescibeCommand("balance-leader-scheduler", "normal", "")
+	checkSchedulerDescribeCommand("balance-leader-scheduler", "normal", "")
 
 	// set label scheduler to disabled manually.
 	echo = mustExec([]string{"-u", pdAddr, "scheduler", "add", "label-scheduler"}, nil)

